osu.Game/Screens/Select/Leaderboards/Leaderboard.cs

```diff
@@ -109,33 +179,120 @@ public BeatmapInfo Beatmap
         }
 
         [BackgroundDependencyLoader(permitNulls: true)]
-        private void load(APIAccess api)
+        private void load(APIAccess api, OsuGame osuGame)
         {
             this.api = api;
+            this.osuGame = osuGame;
+
+            if (osuGame != null)
+                osuGame.Ruleset.ValueChanged += handleRulesetChange;

```

Comment:
For bindables, rather than binding/unbinding to global bindables, you should create a local bindable `private readonly Bindable<Ruleset> ruleset = new Bindable<Ruleset>(); ` and use `ruleset.BindTo(osuGame.Ruleset)`

Reply:
you don't need to unbind `ValueChanged` if you do this.

for the api unfortunately you will have to do the unbind because `OnStateChange` is not a bindable currently.